### ISSUE TITLE
fetchDevices does not return devices name

### ISSUE THREAD

**Issue by gianluigi-bresaola** [aws-amplify/amplify-android#2701]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
// Put output below this line
implementation 'com.amplifyframework:core:2.14.9'
implementation 'com.amplifyframework:aws-auth-cognito:2.14.9'

```

</details>

### Environment information

<details>

```
# Put output below this line
------------------------------------------------------------         
Gradle 7.5.1                                                         
------------------------------------------------------------         
                                                                     
Build time:   2022-08-05 21:17:56 UTC                                
Revision:     d1daa0cbf1a0103000b71484e1dbfe096e095918               
                                                                     
Kotlin:       1.6.21                                                 
Groovy:       3.0.10                                                 
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.7 (JetBrains s.r.o. 17.0.7+0-b2043.56-10550314)   
OS:           Windows 10 10.0 amd64 

```

</details>

### Describe the bug

After succefully login I call fetchDevices to know name and id of each device remembered, but I found "name" field is allways null.
I put a breakpoint in the source code , in the RealAWSCognitoAuthPlugin class, line 1335, and I found that device name is not null, but is not set as a result.
Here is the snippet from source code
val _devices = response?.devices
                val authdeviceList = mutableListOf<AuthDevice>()
                _devices?.forEach {
                    authdeviceList.add(AuthDevice.fromId(it.deviceKey ?: ""))
                }
                onSuccess.accept(authdeviceList)

### Reproduction steps (if applicable)

1) login on one or more devices
2) call rememberDevice function on each one
3) call fetchDevices after login on one device

### Code Snippet

```java
// Put your code below this line.

```
Amplify.Auth.fetchDevices({
                              },{
})

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by mattcreaser** [comment]
@gianluigi-bresaola thanks for your report.

I've confirmed that Amplify Android also didn't set the device name [in v1](https://github.com/aws-amplify/amplify-android/blob/v1/aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java#L687), but that Amplify does return the device name on other platforms in the attributes (see [Swift](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Service/Helpers/ListDevicesOutputResponse%2BHelper.swift#L15) and [Flutter](https://github.com/aws-amplify/amplify-flutter/blob/8c256951464615e9bd72001f6e6b808b01e85033/packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart#L1022)).

Amplify Android does not expose the device attributes, so we can either add the attributes to `AuthDevice` or pull out the `device_name` attribute to set the `name` property, or both.

**Comment by mattcreaser** [comment]
This has been released in Amplify 2.14.10 🎉
